cpp/cmake_modules/BuildUtils.cmake

```diff
@@ -375,6 +375,14 @@ function(ADD_ARROW_LIB LIB_NAME)
                           LINK_PRIVATE
                           ${ARG_SHARED_PRIVATE_LINK_LIBS})
 
+    if(USE_OBJLIB)
+      foreach(ARG_SHARED_LINK_LIB ${ARG_SHARED_LINK_LIBS})
```

Comment:
Minor, but naming `ARG_SOMETHING` a variable that is not one of the original function arguments is confusing.

Reply:
eb88cfd64